File: server/src/main/kotlin/com/coooolfan/uniboard/controller/ProbeScriptController.kt
package com.coooolfan.uniboard.controller

import com.coooolfan.uniboard.service.ProbeScriptService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/probe-script")
class ProbeScriptController(private val probeScriptService: ProbeScriptService) {

    /**
     * 获取自定义安装脚本
     *
     * @param probeId 探针ID
     * @param key 探针密钥
     * @param interval 采集间隔（秒）
     * @return 安装脚本内容
     */
    @GetMapping("/installer/probe/{probeId}/key/{key}/interval/{interval}")
    fun getCustomInstallScript(
        @PathVariable probeId: Long,
        @PathVariable key: String,
        @PathVariable interval: Int,
    ): String {
        return probeScriptService.genCustomInstallScript(probeId, key, interval)
    }


    /**
     * 获取监控脚本
     *
     * @return 监控脚本内容
     */
    @GetMapping("/probe-monitor.sh", produces = ["text/plain"])
    fun getMonitorScript(): String {
        return probeScriptService.genMonitorScript()
    }
}

File: web/src/__generated/services/ProbeScriptController.ts
import type {Executor} from '../';

export class ProbeScriptController {
    
    constructor(private executor: Executor) {}
    
    /**
     * 获取自定义安装脚本
     * 
     * @parameter {ProbeScriptControllerOptions['getCustomInstallScript']} options
     * - probeId 探针ID
     * - key 探针密钥
     * - interval 采集间隔（秒）
     * @return 安装脚本内容
     */
    readonly getCustomInstallScript: (options: ProbeScriptControllerOptions['getCustomInstallScript']) => Promise<
        string
    > = async(options) => {
        let _uri = '/api/probe-script/installer/probe/';
        _uri += encodeURIComponent(options.probeId);
        _uri += '/key/';
        _uri += encodeURIComponent(options.key);
        _uri += '/interval/';
        _uri += encodeURIComponent(options.interval);
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<string>;
    }
    
    /**
     * 获取监控脚本
     * 
     * @return 监控脚本内容
     */
    readonly getMonitorScript: () => Promise<
        string
    > = async() => {
        let _uri = '/api/probe-script/probe-monitor.sh';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<string>;
    }
}

export type ProbeScriptControllerOptions = {
    'getCustomInstallScript': {
        /**
         * 探针ID
         */
        readonly probeId: number, 
        /**
         * 探针密钥
         */
        readonly key: string, 
        /**
         * 采集间隔（秒）
         */
        readonly interval: number
    }, 
    'getMonitorScript': {}
}
